### DISCUSSION TITLE
Alternative to wrap incremental button?

### DISCUSSION THREAD

**Discussion by tompad2** [helgoboss/helgobox#1167] [Q&A] [answered]
Hi!

I have a button on controller that control a switch on a plugin with 3 modes - A,B and C.
As it is now the button switches thru the different modes and when it comes to C it
wraps back to A. Is it possible to make the button go A - B - C and then back to B - A
and turn around again, like A B C B A B C B etc with every button click? How?

**Comment by helgoboss** [comment, marked as answer]
Yes. There are many ways in fact.

One of the easiest is using a control transformation:

```
// You should adjust this
count = 4;
v[0] = 0.1;
v[1] = 0.3;
v[2] = 0.5;
v[3] = 0.3;

// Leave as is
x > 0 ? (
    i += 1;
    y = v[i % count];
);
```

Alternatives:

- **Target "ReaLearn: Browse group mappings"** (each mapping in the group would then represent one possible value)
- **Value sequence** (would be the easiest way if you used absolute control, but since you want relative control via incremental buttons, it's not so straightforward)

**Reply by tompad2** [reply]
I can live with a control transformation :-)
Thanks Benjamin!

**Reply by tompad2** [reply]
Lovely!
Control transformation is da shit! :-)
